astropy/_dev/_scm_version.py

```diff
@@ -0,0 +1,8 @@
+# Try to use setuptools_scm to get the current version; this is only used
```

Comment:
I had a thought about this overnight. Is there a reason why this needs to be a submodule rather than a single file in the root of the repo? It feels like that would be less obvious and prone to confusing people?

Reply:
I was wondering that too and thought that if it’s not in the package then there is a non zero chance that file could exist one level up in an installed version of astropy?

Reply:
did you mean to say "if it's in the package"? otherwise I am not sure I understand what you are saying.

Reply:
I meant that if ``_scm_version.py`` is at the root of the repo, and we check for ``../_scm_version.py`` in ``version.py``, then if there happens to be a ``_scm_version.py`` file in ``site-packages``, one could run into issues. Having the file be inside the package somewhat guarantees there won't be collisions with other files in the site-packages directory?

Reply:
Maybe have it as `./_scm_version.py`, i.e., same directory as `version.py`? It would seem a bit cleaner.

Reply:
> if there happens to be a _scm_version.py file in site-packages

oh yeah you are correct this will trigger the original

Reply:
Having it in the astropy namespace ensures it's always the astropy one and not another one getting picked up via relative import. That being said, it could just be a file in the astropy folder not a folder?

Reply:
Yes I agree, maybe no need for the _dev namespace

Reply:
Originally I had it in `astropy/_scm_version.py` but unless you can tell me a way to exclude a single file from wheels (see https://github.com/pypa/setuptools/issues/511) this has to be an entire sub-package.  In any case there might be other uses found for it.

One thing I should do is rename it from `astropy._dev._scm_version` to `astropy._dev.scm_version`.  Since the subpackage is already "private" there's no reason for the module name to be as well.

Reply:
Ah, that's an oddity!  So, seems fine to stick to what you have then!